### PR TITLE
RESTEasy Reactive docs

### PR DESCRIPTION
**Pull request by FroMage**
Upstream PR has been merged at https://github.com/quarkusio/quarkus/pull/13802

### CODE REVIEW

File: _data/guides.yaml [hunk 1]
```diff
@@ -99,6 +99,9 @@ categories:
       - title: Consuming and Implementing GraphQL Services
         url: /guides/microprofile-graphql
         description: This guide explains how to leverage Eclipse MicroProfile GraphQL to consume and implement GraphQL services.
+      - title: RESTEasy Reactive
+        url: /guides/rest-reactive
```
Comment (gsmet) [inline _data/guides.yaml:103]: I think it would be better for SEO if the URL was `resteasy-reactive`.

Comment (FroMage) [reply _data/guides.yaml:103]: Probably. I'm mostly putting this up to help the review of https://github.com/quarkusio/quarkus/pull/13802 so better you wait for @geoand and @stuartwdouglas 's first pass before reviewing this, the paint's still fresh :)

**Comment by FroMage** [comment]
Squashed and ready for review, though we do have to wait for an official OK about the JAX-RS spec publication, so don't merge this.

**Comment by FroMage** [comment]
@gsmet can we merge this to get it on the website for people to try out or do we want for a release?

**Pull request merged**